### PR TITLE
Print alphabetically ordered summary list output

### PR DESCRIPTION
**Pull request by kloczek**
To be able easier status of some detection I thonk that it is
better to keep alphabvetically ordered summary list of detected
features.

Signed-off-by: Tomasz Kłoczko <kloczek@github.com>

## What does this PR do?
Prints better readable configure summary list
## What are related issues/pull requests?
None.
## Tasklist

 - [ ] ADD YOUR TASKS HERE
 - [ ] Add test case(s)
 - [ ] Add documentation
 - [ ] Review
 - [ ] Adjust for comments
 - [ ] All CI builds and checks have passed

### CODE REVIEW

**Comment by kloczek** [comment]
Only with that patch I was able to spot somee contradicting output.
```console
  HDF4 support:              no
  HDF5 support:              yes
  HDFS support:              no
```
Those three lines are printed by:
```
LOC_MSG([  HDF4 support:              ${HAVE_HDF4}])
LOC_MSG([  HDF5 support:              ${HAVE_HDF5}])
LOC_MSG([  HDFS support:              ${HDFS_ENABLED}])
```
I thoink that those HDF5 lines should be a bit modified to provide understandable resuls but that is sometbing which could be added after that patch.

File: gdal/configure.ac [hunk 1]
```diff
@@ -6064,94 +6064,90 @@ LOC_MSG([  C++14 support:             ${CXX14_SUPPORT}])
...
-fi
-LOC_MSG([  Kakadu support:            ${HAVE_KAKADU}])
 if test "x$HAVE_JASPER_UUID" != "x" ; then
   LOC_MSG([  JasPer support:            ${HAVE_JASPER} (GeoJP2=${HAVE_JASPER_UUID})])
 else
   LOC_MSG([  JasPer support:            ${HAVE_JASPER}])
 fi
-LOC_MSG([  OpenJPEG support:          ${HAVE_OPENJPEG}])
+LOC_MSG([  JPEG 12 bit:               ${JPEG12_ENABLED}])
+LOC_MSG([  JPEG-in-TIFF 12 bit:       ${TIFF_JPEG12_ENABLED}])
```
Comment (rouault) [inline gdal/configure.ac:6073]: this entry and the 2 above ones aren't alphabetically sorted

File: gdal/configure.ac [hunk 2]
```diff
@@ -6064,94 +6064,90 @@ LOC_MSG([  C++14 support:             ${CXX14_SUPPORT}])
...
-LOC_MSG([  HEIF support:              ${HAVE_HEIF}])
 LOC_MSG([  userfaultfd support:       ${ENABLE_UFFD}])
-LOC_MSG([  misc. gdal formats:       ${GDALFORMATS_ENABLED}])
-LOC_MSG([  disabled gdal formats:    ${GDALFORMATS_DISABLED}])
-LOC_MSG([  misc. ogr formats:        ${OGRFORMATS_ENABLED}])
-LOC_MSG([  disabled ogr formats:     ${OGRFORMATS_DISABLED}])
+LOC_MSG([  WebP support:              ${WEBP_SETTING}])
+LOC_MSG([  Xerces-C support:          ${HAVE_XERCES}])
+LOC_MSG([  ZSTD support:              ${ZSTD_SETTING}])
+LOC_MSG([  Kakadu support:            ${HAVE_KAKADU}])
```
Comment (rouault) [inline gdal/configure.ac:6150]: Kakadu not correctly ordered

**Review by rouault — COMMENTED** [review]
Please also commit the regenerated configure file

**Comment by kloczek** [comment]
> Please also commit the regenerated configure file

All autoconf, automake, autoheader and libtool files should not be present in git repo.
To genberate all thise files all what is needed is just tun `autoreconf -fiv`.
I've tested everything  under autoconf 2.71 and autoreconf even with that version generates corrct build framework.

**Comment by rouault** [comment]
> All autoconf, automake, autoheader and libtool files should not be present in git repo.

I know. This is tracked at #3561. But that's how things are currently, so that CI can catch configure.ac changes since it doesn't run autoconf

**Comment by kloczek** [comment]
Testing procedure which I've been using consif from `autoreconf -fiv; ./configure;  make distcheck`
`distcheck` always should be used to generate dist tar bal because that target not only generates dist tar ball but tries to build something out of that tar ball.

**Comment by kloczek** [comment]
Thank you :)

**Pull request merged**